### DISCUSSION TITLE
VS Code linker error on Mac OS

### DISCUSSION THREAD

**Discussion by jnaraja** [open-source-parsers/jsoncpp#1387] [Q&A]
Hello there,

I have been trying to figure out how to solve a linker error but I have had no luck. I read in the examples folder that I could perform this command:

g++ stringWrite.cpp -ljsoncpp -std=c++11 -D_GLIBCXX_USE_CXX11_ABI=0 -o stringWrite

but I receive this error:

ld: library not found for -ljsoncpp
clang: error: linker command failed with exit code 1 (use -v to see invocation)

If anyone could help, I would greatly appreciate it!

**Comment by BillyDonahue** [comment]
Try adding the --verbose flag to g++ to see what's in the g++ LIBRARY_PATH list.
libjsoncpp.a would have to be in one of those paths for this to work.
If you need to add a new library path to find jsoncpp you can do it with the -L flag.

All this nonsense is handled by pkg-config, which is a tool that cooks up the right include and link command options for you. https://people.freedesktop.org/~dbn/pkg-config-guide.html  jsoncpp does provide a pkg-config/jsoncpp.pc in its installation for this purpose.
